server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -401,13 +402,12 @@ public String executor() {
                 new InitializeExtensionRequest(transportService.getLocalNode(), extension),
                 initializeExtensionResponseHandler
             );
-            // TODO: make asynchronous
-            inProgressFuture.get(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+            inProgressFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
         } catch (Exception e) {
-            try {
-                throw e;
-            } catch (Exception e1) {
-                logger.error(e.toString());
+            if (e.getCause() instanceof TimeoutException) {
+                logger.error("Timed out awaiting for the response from the extension {}", extension);
+            } else {
+                logger.error(e.getCause().getMessage());
```

Comment:
The cause could be `null` (since you are catching any exception), please check for that